### PR TITLE
[node] - Add query type to URL

### PR DESCRIPTION
**Pull request by harryadel**
I'm using query property but getting an error telling me it doesn't exist.

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/http.html#http_class_http_incomingmessage

### CODE REVIEW

**Review by ryanrw — APPROVED** [review]

**Review by Flarna — CHANGES_REQUESTED** [review]
Could you please add a link to the documentation (or source code if it is not documented) where this property is set?

Additionally this requires a test

File: types/node/http.d.ts [hunk 1]
```diff
@@ -334,6 +334,7 @@ declare module "http" {
          * Only valid for request obtained from http.Server.
          */
         url?: string;
+        query?: {[key: string]: any}
```
Comment (Flarna) [inline types/node/http.d.ts:337]:
```suggestion
        query?: {[key: string]: unknown}
```

**Comment by harryadel** [comment]
> Could you please add a link to the documentation (or source code if it is not documented) where this property is set?
> 
> Additionally this requires a test

https://github.com/nodejs/node/blob/7efada695f9cb7515a78565213f25d8077c206ff/lib/url.js#L67
And I'll add the test right now

**Comment by Flarna** [comment]
Your link points to url but you add query to `http.IncomingMessage`. 

url has already a `query` property, see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/769bc304622fa48d26edaa485191e678d01ea2e1/types/node/url.d.ts#L32

**Comment by harryadel** [comment]
> Your link points to url but you add query to `http.IncomingMessage`.
> 
> url has already a `query` property, see
> 
> https://github.com/DefinitelyTyped/DefinitelyTyped/blob/769bc304622fa48d26edaa485191e678d01ea2e1/types/node/url.d.ts#L32

hmm, My issues stems from this block of code:

```
WebApp.connectHandlers.use(async (httpReq, httpResp, next) => {
   
   httpReq.query <--
   
```      
Which results in 
```
Property 'query' does not exist on type 'IncomingMessage'.
```

**Comment by Flarna** [comment]
`WebApp.connectHandlers.use()` seems to be some framework on top of nodejs http. Most likely the type definition from this framework needs some improvements as it adds properties on set by node itself.

**Comment by harryadel** [comment]
https://github.com/meteor/meteor/blob/devel/packages/webapp/webapp_server.js#L892
It's the framework issue indeed, sorry for the inconvenience.

**Review by SimonSchick — CHANGES_REQUESTED** [review]
Nothing on the docs points to an existance of this property.

**Pull request closed without merging**